src/refactor_bot/agents/planner.py

```
@@ -119,19 +268,52 @@ def decompose(
-            raise PlanningError(f"Failed to call Claude API: {e}") from e
+        # Step 4: Call LLM API with tool-use
+        response = None
+        used_provider: str = ""
+        providers = self._provider_chain()
+        last_error: Exception | None = None
+        tool_schema = self._get_tool_schema()
+        for index, provider in enumerate(providers):
+            try:
+                if provider == "anthropic":
+                    if not self._anthropic_client:
+                        raise PlanningError("Anthropic client unavailable")
```

Ruff (0.15.1)

[warning] Abstract `raise` to an inner function

(TRY301)